backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/user/api/impl/UserDAO.java

```diff
@@ -312,6 +314,19 @@ public void save(@NonNull final org.compiere.model.I_AD_User user)
 		return Optional.of(getById(targetUserId));
 	}
 
+	@Override
+	public List<UserId> retrieveUsersByJobId(@NonNull final JobId jobId)
+	{
+		return Services.get(IQueryBL.class)
+				.createQueryBuilder(I_AD_User.class)
+				.addOnlyActiveRecordsFilter()
+				.addEqualsFilter(I_AD_User.COLUMNNAME_C_Job_ID, jobId)
+				.create()
+				.stream()
+				.map(adUser -> UserId.ofRepoId(adUser.getAD_User_ID()))
+				.collect(ImmutableList.toImmutableList());
```

Comment:
```suggestion
				.listIds(UserId::ofRepoId);
```

Avoid loading all the records in order to just get their IDs
Note: this returns an `ImmutableSet<UserId>`.